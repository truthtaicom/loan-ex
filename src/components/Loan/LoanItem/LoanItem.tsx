import React, { useCallback } from 'react';
import { List, Avatar, Skeleton, Button} from 'antd'

export interface LoanItemProps {
  amount: number;
  id: string;
  name: string;
  loanTerm: number;
  payWeekly: number;
  createdAt: string;
  owner: string;
  paid: boolean;
}

export function PayButton(props) {
  const { id, payNow } = props;
  const onPayNow = useCallback(() => {
    payNow(id)
  }, [id, payNow])

  if(props.paid) {
    return <Button type="danger" onClick={onPayNow}>Re-pay</Button>
  }

  return <Button type="primary" onClick={onPayNow}>Pay now</Button>
}

export interface LoanItemExtraProps {
  loading: boolean,
  payNow: (id: string) => void
}

export default function LoanItem(props: LoanItemProps & LoanItemExtraProps) {
  return (
    <List.Item actions={[<PayButton {...props} />]}>
      <Skeleton avatar title={true} loading={props.loading} active>
        <List.Item.Meta
          avatar={<Avatar style={{ backgroundColor: '#87d068' }}>{props.name[0]}</Avatar>}
          title={props.name}
          description={`Loan term: ${props.loanTerm} month`}
        />
      </Skeleton>
    </List.Item>
  )
}

