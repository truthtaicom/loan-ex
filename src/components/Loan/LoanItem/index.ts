export { default as LoanItem } from './LoanItem';
