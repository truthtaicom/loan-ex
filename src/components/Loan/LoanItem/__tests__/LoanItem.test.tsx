import React from 'react';
import { render } from '@testing-library/react';
import LoanItem, { PayButton } from '../LoanItem';
import loanMock from '../../../../services/loan.json'

describe('<LoanItem /> spec', () => {
  const payNow = () => jest.fn()
  it('renders LoanItem component', () => {
    const component = render(<LoanItem {...loanMock[0]} loading payNow={payNow} />);
    expect(component).toMatchSnapshot();
  });

  it('renders PayButton component with re-pay', () => {
    const component = render(<PayButton isPaid payNow={payNow} />);
    expect(component).toMatchSnapshot();
  });

  it('renders PayButton component with pay now', () => {
    const component = render(<PayButton payNow={payNow} />);
    expect(component).toMatchSnapshot();
  });
});
