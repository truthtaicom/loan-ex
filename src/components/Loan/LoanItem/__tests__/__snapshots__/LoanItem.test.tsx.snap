// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoanItem /> spec renders LoanItem component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="ant-list-item"
      >
        <div
          class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
            </ul>
          </div>
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Pay now
              </span>
            </button>
          </li>
        </ul>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="ant-list-item"
    >
      <div
        class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
      >
        <div
          class="ant-skeleton-header"
        >
          <span
            class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
          />
        </div>
        <div
          class="ant-skeleton-content"
        >
          <h3
            class="ant-skeleton-title"
            style="width: 50%;"
          />
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
          </ul>
        </div>
      </div>
      <ul
        class="ant-list-item-action"
      >
        <li>
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            <span>
              Pay now
            </span>
          </button>
        </li>
      </ul>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoanItem /> spec renders PayButton component with pay now 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="ant-list-item"
      >
        <div
          class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
            </ul>
          </div>
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Pay now
              </span>
            </button>
          </li>
        </ul>
      </li>
    </div>
    <div>
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Pay now
        </span>
      </button>
    </div>
    <div>
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Pay now
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        Pay now
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LoanItem /> spec renders PayButton component with re-pay 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="ant-list-item"
      >
        <div
          class="ant-skeleton ant-skeleton-with-avatar ant-skeleton-active"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
            </ul>
          </div>
        </div>
        <ul
          class="ant-list-item-action"
        >
          <li>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Pay now
              </span>
            </button>
          </li>
        </ul>
      </li>
    </div>
    <div>
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Pay now
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        Pay now
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
